addon/waiter-manager.ts

```diff
@@ -52,6 +52,7 @@ export function getWaiters(): IWaiter[] {
  * @public
  */
 export function _reset(): void {
+  WAITERS.forEach(waiter => ((waiter as any).isRegistered = false));
```

Comment:
This would be good in a separate PR. I can see why you're doing this; to ensure all waiter references have their properties set to `false`. It seems like a good edition, but I'd just prefer it's added outside of this PR as it muddies the waters.

Reply:
This is actually required to make the tests pass. The issue was that the waiters would never re-register themselves once this was called because of [this check](https://github.com/rwjblue/ember-test-waiters/blob/873f7d577c34c47f33601b628e59564451a7c53d/addon/test-waiter.ts#L46). Previously only [one test](https://github.com/rwjblue/ember-test-waiters/blob/873f7d577c34c47f33601b628e59564451a7c53d/tests/unit/wait-for-promise-test.ts#L23-L33) was verifying that the promise waiter was actually waiting (and the [other two](https://github.com/rwjblue/ember-test-waiters/blob/873f7d577c34c47f33601b628e59564451a7c53d/tests/unit/wait-for-promise-test.ts#L40-L62) weren't actually testing anything because the waiter wasn't get re-registered after the first test). My changes cause multiple tests to verify that the waiter is installed, and all but the first will fail without this change.

So I guess I can break this out into a separate PR that this one would depend on if you like, but that's the reason that it's in this PR.

Reply:
Hmmm ok. This does seem like a bug, as the waiters state wouldn't be reset in all the references if we didn't do this. I think this is a good catch. 

The reason I keep asking to split this apart into separate PRs is this one is quite large, and does seem to span a few different types of changes - blueprint updates, node version changes, bugfixes, and feature additions. It's just a bit hard to parse the whole thing. 

I'll leave it up to you. I won't die if this stays in.

Reply:
Heh, I really don't feel strongly here. I'll tell you my, probably over-pedantic, hesitation to break it out into a separate PR (warning, some insight into the sometimes-silly inner workings of my brain ahead).

I really don't like committing code without test coverage, but since this, as far as I can tell, only exists for test instrumentation, I hesitate to write a test to test the test instrumentation. But maybe that's the right way to go? Or maybe just PR it without test coverage?

When it's part of this PR, I feel better because there is actually a test that will fail if this code doesn't exist (even though it's not explicitly testing the instrumentation...).

So I end up with decision paralysis, and would by default just leave it in, but can be swayed by the tiniest nudge 😜

Reply:
Ya I'm actually with you on the 'not testing the testing impl'. Let's keep it here. Done!